common/src/proto.rs

```diff
@@ -54,9 +48,15 @@ pub struct Spawns {
 #[derive(Debug, Serialize, Deserialize)]
 pub struct Command {
     pub generation: u16,
+    pub player_input: CharacterInput,
     pub orientation: na::UnitQuaternion<f32>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CharacterInput {
     /// Relative to the character's current position, excluding orientation
-    pub velocity: na::Vector3<f32>,
+    pub movement: na::Vector3<f32>,
+    pub no_clip: bool,
```

Comment:
It's strange for this to be added with no logic that actually depends on it in the same commit.

Reply:
I was unsure whether to put these changes in the first or second commit, so thanks for this feedback!